src/Sylius/Bundle/ResourceBundle/DependencyInjection/Driver/AbstractDriver.php

```diff
@@ -0,0 +1,193 @@
+        }
+    }
+
+    /**
+     * @param ContainerBuilder $container
+     * @param MetadataInterface $metadata
+     */
+    protected function addController(ContainerBuilder $container, MetadataInterface $metadata)
+    {
+        // @todo: Remove when ResourceController is reworked.
+        $configurationDefinition = new Definition(new Parameter('sylius.controller.configuration.class'));
+        $configurationDefinition
+            ->setFactory(array(new Reference('sylius.controller.configuration_factory'), 'createConfiguration'))
+            ->setArguments(array($metadata->getApplicationName(), $metadata->getName(), $metadata->getTemplatesNamespace()))
+            ->setPublic(false)
+        ;
+
+        $definition = new Definition($metadata->getClass('controller'));
+        $definition
+            ->setArguments(array($configurationDefinition))
+            ->addMethodCall('setContainer', array(new Reference('service_container')))
+        ;
+
+        $container->setDefinition($metadata->getServiceId('controller'), $definition);
+    }
+
+    /**
+     * @param ContainerBuilder $container
+     * @param MetadataInterface $metadata
+     */
+    protected function addFactory(ContainerBuilder $container, MetadataInterface $metadata)
+    {
+        $translatableFactoryInterface = 'Sylius\Component\Translation\Factory\TranslatableFactoryInterface';
+
+        $factoryClass = $metadata->getClass('factory');
+        $modelClass = $metadata->getClass('model');
+
+        $reflection = new \ReflectionClass($factoryClass);
+        $definition = new Definition($factoryClass);
+
+        if (interface_exists($translatableFactoryInterface) && $reflection->implementsInterface($translatableFactoryInterface)) {
+            $decoratedDefinition = new Definition(Factory::class);
+            $decoratedDefinition->setArguments(array($modelClass));
+
+            $definition->setArguments(array($decoratedDefinition, new Reference('sylius.translation.locale_provider')));
+
+            $container->setDefinition($metadata->getServiceId('factory'), $definition);
+
+            return;
+        }
+
+        $definition->setArguments(array($modelClass));
+
+        $container->setDefinition($metadata->getServiceId('factory'), $definition);
+    }
+
+    /**
+     * @param ContainerBuilder $container
+     * @param MetadataInterface $metadata
+     */
+    protected function addForms(ContainerBuilder $container, MetadataInterface $metadata)
+    {
+        foreach ($metadata->getClass('form') as $formName => $formClass) {
+            $suffix = 'default' === $formName ? '' : sprintf('_%s', $formName);
+            $alias = sprintf('%s_%s%s', $metadata->getApplicationName(), $metadata->getName(), $suffix);
+
+            $definition = new Definition($formClass);
+
+            switch ($formName) {
+                case 'choice':
+                    $definition->setArguments(array(
```

Comment:
This is existing behavior and I did not want to change it. Not sure if it makes sense, but let's leave that for a separate RFC.